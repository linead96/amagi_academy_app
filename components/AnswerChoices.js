import React from 'react';
import {
  View,
  TouchableNativeFeedback,
  StyleSheet,
  } from 'react-native'
import { Icon } from 'expo';

const AnswerChoices = (props) => {
  return props.flipped ? 
    (<View style={styles.answerKeyChoicesContainer}>
      <TouchableNativeFeedback>
        <View style={styles.answerKeyChoicesCorrect}>
              <Icon.Ionicons
                name={'md-checkmark'}
                size={50}
                color={'white'}
              />
        </View>
      </TouchableNativeFeedback>
      <TouchableNativeFeedback>
        <View style={styles.answerKeyChoicesWrong}>
              <Icon.Ionicons
                name={'md-close'}
                size={50}
                color={'white'}
              />
        </View>
      </TouchableNativeFeedback>
    </View>
    ):null
}

const styles = StyleSheet.create({
    answerKeyChoicesContainer: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
      },
      answerKeyChoicesCorrect: {
        backgroundColor: '#89E171',  
        flex: 1,
        margin: '5%',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 25,
      },
      answerKeyChoicesWrong: {
        backgroundColor: '#F55241',  
        flex: 1,
        margin: '5%',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 25,
      },
});

export default AnswerChoices;
