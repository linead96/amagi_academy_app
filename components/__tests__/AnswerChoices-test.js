import 'react-native';
import React from 'react';
import AnswerChoices from '../AnswerChoices';
import renderer from 'react-test-renderer';

test('renders two view with icon when the flashcard component state flipped is true', () => {
    const tree = renderer.create(<AnswerChoices flipped={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders null when the flashcard component state flipped is false', () => {
    const tree = renderer.create(<AnswerChoices flipped={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});