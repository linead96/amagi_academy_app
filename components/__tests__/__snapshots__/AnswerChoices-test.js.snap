// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders null when the flashcard component state flipped is false 1`] = `null`;

exports[`renders two view with icon when the flashcard component state flipped is true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;
